src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1361 @@
+                $this->switchToWindow($origName);
+            }
+        }
+    }
+
+    /**
+     * @throws DriverException
+     */
+    private function findElement(
+        #[Language('XPath')]
+        string $xpath,
+        RemoteWebElement $parent = null
+    ): RemoteWebElement {
+        try {
+            $finder = WebDriverBy::xpath($xpath);
+            return $parent
+                ? $parent->findElement($finder)
+                : $this->getWebDriver()->findElement($finder);
+        } catch (Throwable $e) {
+            throw new DriverException("Failed to find element: {$e->getMessage()}", 0, $this->wrapError($e));
+        }
+    }
+
+    /**
+     * @throws DriverException
+     */
+    private function selectRadioValue(RemoteWebElement $element, string $value): void
+    {
+        // short-circuit when we already have the right button of the group to avoid XPath queries
+        if ($element->getAttribute('value') === $value) {
+            $element->click();
+
+            return;
+        }
+
+        $name = $element->getAttribute('name');
+        if (!$name) {
+            throw new DriverException(sprintf('The radio button does not have the value "%s"', $value));
+        }
+        if ($name === true) {
+            $name = '';
+        }
+
+        $formId = $element->getAttribute('form');
+        if ($formId === true) {
+            $formId = '';
+        }
+
+        try {
+            $escapedName = $this->xpathEscaper->escapeLiteral($name);
+            $escapedValue = $this->xpathEscaper->escapeLiteral($value);
+            if (null !== $formId) {
+                $escapedFormId = $this->xpathEscaper->escapeLiteral($formId);
+                $input = $this->findElement(
+                    <<<"XPATH"
+                    //form[@id=$escapedFormId]//input[@type="radio" and not(@form) and @name=$escapedName and @value=$escapedValue]
+                    |
+                    //input[@type="radio" and @form=$escapedFormId and @name=$escapedName and @value=$escapedValue]
+                    XPATH
+                );
+            } else {
+                $input = $this->findElement(
+                    "./ancestor::form//input[@type=\"radio\" and not(@form) and @name=$escapedName and @value=$escapedValue]",
+                    $element
+                );
+            }
+        } catch (DriverException $e) {
+            $message = sprintf('The radio group "%s" does not have an option "%s"', $name, $value);
+            throw new DriverException($message, 0, $e);
+        }
+
+        $input->click();
+    }
+
+    /**
+     * @throws DriverException
+     */
+    private function selectOptionOnElement(RemoteWebElement $element, string $value, bool $multiple = false): void
+    {
+        $escapedValue = $this->xpathEscaper->escapeLiteral($value);
+        // The value of an option is the normalized version of its text when it has no value attribute
+        $optionQuery = sprintf(
```

Comment:
I suggest using `WebDriverSelect` instead of redoing a similar logic, if possible.